packages/dds/tree/src/test/shared-tree/sharedTree.spec.ts

```diff
@@ -1090,6 +1094,210 @@ describe("SharedTree", () => {
 				});
 			}
 		});
+
+		describe("can resubmit", () => {
+			const sb = new SchemaBuilder({ scope: "shared tree undo tests" });
+			const schema = sb.intoSchema(sb.list(sb.list(sb.string)));
+
+			/**
+			 * Each test will involve two peers, one of which has removed a tree (and is therefore able to restore it) -
+			 * let's call that one peer 1.
+			 *
+			 * Peer 1 can do one of three things:
+			 * (A) Commit 1: make an edit to the removed tree, Commit 2: make another edit to removed tree
+			 * (B) Commit 1: make an edit to the removed tree, Commit 2: restore the removed tree
+			 * (C) Commit 1: restore the removed tree, Commit 2: make an edit to the restored tree
+			 * Peer 2 can only do option (A) above.
+			 */
+			const scenarios = [
+				["A", "A"],
+				["A", "B"],
+				["A", "C"],
+				["B", "A"],
+				["C", "A"],
+			];
+
+			function makeTestTitle(peer: string): string {
+				switch (peer) {
+					case "A": {
+						return "an edit to the removed tree after another edit";
+					}
+					case "B": {
+						return "the restoration of a tree after an edit to the tree";
+					}
+					case "C": {
+						return "the edit to a removed tree after its restoration";
+					}
+					default:
+						fail("invalid test case");
+					// unreachableCase(peer);
```

Comment:
this isn't working cause "type string can't be assigned to type never" but this seems to be how other usages of unreachableCase work, any ideas what's going on?

Reply:
you may need to type `peer` with `"A" | "B" | "C"`

Reply:
I ended up just going with fail cause doing this requires me to cast the inputs as well

Reply:
I think you'd be better off using an enum. It will make the code more self-documenting (also "A", "B", "C" is not supper clear) and you'll be able to use the `unreachableCase `.